### ISSUE TITLE
unexpected Applicative[Set] conjured by .sequenceU

### ISSUE THREAD

**Issue by tpolecat** [scalaz/scalaz#908]
I was surprised this compiled, and then was surprised at the result.

``` scala
scala> List(Set(1,2,3)).sequenceU
res23: scala.collection.immutable.Set[Int] = Set(1, 2, 3)
```

because

``` scala
scala> reflect.runtime.universe.reify { List(Set(1,2,3)).sequenceU }
res25: reflect.runtime.universe.Expr[scala.collection.immutable.Set[Int]] = 
Expr[scala.collection.immutable.Set[Int]](
  Scalaz.ToTraverseOps(List.apply(Predef.Set.apply(1, 2, 3)))(Scalaz.listInstance)
    .sequenceU(Unapply.unapplyA(Applicative.monoidApplicative(Scalaz.setMonoid))))
```

Is this a feature? 

/cc @adelbertc

**Comment by adelbertc** [comment]
Related:

Applicative.monoidApplicative: https://github.com/scalaz/scalaz/blob/3d270716549f8a3edb419b6c592a1dbb4c50723d/core/src/main/scala/scalaz/Applicative.scala#L121

Which just delegates to `Monoid[M].applicative`: https://github.com/scalaz/scalaz/blob/3d270716549f8a3edb419b6c592a1dbb4c50723d/core/src/main/scala/scalaz/Monoid.scala#L57-67

**Comment by S11001001** [comment]
I'd like to remove this, but a previous removal was reverted in 186693a3869b9c887452911b6bc51d47038a3ee0.

@purefn why was the reversion necessary?

**Comment by purefn** [comment]
That's a good question. I can't recall and haven't been able to find any
records of it in email or anything.

**Comment by S11001001** [comment]
Fixed by #1313.
